osbuild/testutil/net.py

```diff
@@ -96,3 +96,19 @@ def https_serve_directory(rootdir, certfile, keyfile, simulate_failures=0):
         yield httpd
     finally:
         httpd.shutdown()
+
+@contextlib.contextmanager
+def https_serve_directory_mtls(rootdir, ca_cert, server_cert, server_key, simulate_failures=0):
```

Comment:
(super nitpick/follow) the duplication (that pre-exists) is a bit not-nice, maybe something in the spirit of
```python
diff --git a/osbuild/testutil/net.py b/osbuild/testutil/net.py
index 6c94572c..bb8e2885 100644
--- a/osbuild/testutil/net.py
+++ b/osbuild/testutil/net.py
@@ -62,20 +62,23 @@ class DirHTTPServer(ThreadingHTTPServer):
             request, client_address, self, directory=self.directory)
 
 
-def _httpd(rootdir, port, simulate_failures):
-    return DirHTTPServer(
+def _httpd(rootdir, simulate_failures, ctx=None):
+    port = _get_free_port()
+    httpd = DirHTTPServer(
         ("localhost", port),
         http.server.SimpleHTTPRequestHandler,
         directory=rootdir,
         simulate_failures=simulate_failures,
     )
+    if ctx:
+        httpd.socket = ctx.wrap_socket(httpd.socket, server_side=True)
+    threading.Thread(target=httpd.serve_forever).start()
+    return httpd
 
 
 @contextlib.contextmanager
 def http_serve_directory(rootdir, simulate_failures=0):
-    port = _get_free_port()
-    httpd = _httpd(rootdir, port, simulate_failures)
-    threading.Thread(target=httpd.serve_forever).start()
+    httpd = _httpd(rootdir, simulate_failures)
     try:
         yield httpd
     finally:
@@ -84,30 +87,21 @@ def http_serve_directory(rootdir, simulate_failures=0):
 
 @contextlib.contextmanager
 def https_serve_directory(rootdir, certfile, keyfile, simulate_failures=0):
-    port = _get_free_port()
-    httpd = _httpd(rootdir, port, simulate_failures)
-
     ctx = ssl.create_default_context(ssl.Purpose.CLIENT_AUTH)
     ctx.load_cert_chain(certfile=certfile, keyfile=keyfile)
-    httpd.socket = ctx.wrap_socket(httpd.socket, server_side=True)
-
-    threading.Thread(target=httpd.serve_forever).start()
+    httpd = _httpd(rootdir, simulate_failures, ctx)
     try:
         yield httpd
     finally:
         httpd.shutdown()
 
+
 @contextlib.contextmanager
 def https_serve_directory_mtls(rootdir, ca_cert, server_cert, server_key, simulate_failures=0):
-    port = _get_free_port()
-    httpd = _httpd(rootdir, port, simulate_failures)
-
     ctx = ssl.create_default_context(ssl.Purpose.CLIENT_AUTH, cafile=ca_cert)
     ctx.load_cert_chain(certfile=server_cert, keyfile=server_key)
     ctx.verify_mode = ssl.CERT_REQUIRED
-    httpd.socket = ctx.wrap_socket(httpd.socket, server_side=True)
-
-    threading.Thread(target=httpd.serve_forever).start()
+    httpd = _httpd(rootdir, simulate_failures, ctx)
     try:
         yield httpd
     finally:

```

Reply:
Applied cleanly, thanks.